manifests/init.pp

```diff
@@ -413,7 +413,11 @@
       }
     }
 
-    include pulpcore::plugin::container
+    unless $pulpcore_mirror {
+      class { 'pulpcore::plugin::container':
+        registry_base_url => $foreman::config::apache::_proxy_backend,
```

Comment:
I think it would be cleaner to handle this in the `if $shared_with_foreman_vhost {` tree. Whether it's a mirror is irrelevant: what counts is how we configure Apache.

That is something I feel very strongly about, we need to get rid of that. In the future I want be able to deploy a non-mirror on a server where Foreman doesn't run. We should now be making sure that we don't introduce more blockers to this. Once we have that in place, we can create a separate scenario for this. Add shared storage for `/var/lib/pulp`, external PostgreSQL, Redis and a load balancer and you have a horizontally scalable deployment. I'm certain a few downstream customers would appreciate this.

Similarly, you should be able to do the same thing for mirror Pulp instances to horizontally scale them out if needed.